lib/eslint.js

```diff
@@ -54,18 +54,40 @@ function parseBooleanConfig(string) {
 }
 
 /**
- * Parses a JSON-like config.
- * @param {string} string The string to parse.
- * @returns {Object} Result map object
+ * Parses config from JSON-like comments.
+ * @param {string} comments The comments to parse.
+ * @param {Object} location Start line and column of comments for potential error message.
+ * @param {Object[]} messages The messages queue for potential error message.
+ * @returns {Object} Rules configuration as parsed per comments.
  */
-function parseJsonConfig(string) {
-    var items = {};
-    string = string.replace(/([a-z0-9\-\/]+):/g, "\"$1\":").replace(/(\]|[0-9])\s+(?=")/, "$1,");
+function parseCommentsConfig(comments, location, messages) {
+    var items = {},
+        commentRules = {};
+    comments = comments.replace(/([a-zA-Z0-9\-\/]+):/g, "\"$1\":").replace(/(\]|[0-9])\s+(?=")/, "$1,");
     try {
-        items = JSON.parse("{" + string + "}");
-    } catch(e) { }
 
-    return items;
+        items = JSON.parse("{" + comments + "}");
+
+        Object.keys(items).forEach(function(name) {
```

Comment:
Ah sorry, I can see how my comment on the previous commit was unclear. What I meant was to move the `try-catch` and creation of the error message into `parseJsonConfig()`, so that at this spot, you could just use the result.